### PR TITLE
Filter/MLAgent: Initial draft for ml-agent support in TensorFilter

### PR DESCRIPTION
**Pull request by wooksong**
This patch is an initial draft to bring the ML Agent features into NNS.

Signed-off-by: Wook Song <wook16.song@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: gst/nnstreamer/ml_agent.c [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  if (!scheme || g_strcmp0 (URI_SCHEME, scheme)) {
+    return g_steal_pointer (&ret);
+  }
+
+  uri_hier_part = g_strstr_len (ret, -1, ":");
+  while (*uri_hier_part == ':' || *uri_hier_part == '/') {
+    uri_hier_part++;
+  }
+
+  /** NOTE: Only for the following URI formats to get the file path of
```
Comment (jaeyun-jung) [inline gst/nnstreamer/ml_agent.c:47]: minor: plz fix tag `@note`

**Review by jaeyun-jung — APPROVED** [review]

**Review by songgot — APPROVED** [review]
LGTM

File: gst/nnstreamer/ml_agent.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#include <glib-object.h>
+
+#ifdef ENABLE_ML_AGENTD
+/**
+ * @brief Parse the given URI into the valid file path string
+ */
+const gchar *mlagent_parse_uri_string (const GValue * val);
+#else
+#define mlagent_parse_uri_string g_value_get_string
```
Comment (myungjoo) [inline gst/nnstreamer/ml_agent.h:26]: How about an error ptr or null returning static function? (Linux kernel style)

File: meson.build [hunk 3]
```diff
@@ -325,6 +325,17 @@ if not get_option('mxnet-support').disabled()
   endif
 endif
 
+# ml-agentd
+ml_agentd_dep = dependency('', required: false)
```
Comment (myungjoo) [inline meson.build:329]: Why do you need this much complicated? Don't we have pkgconfig enabled for magenta?

Comment (wooksong) [reply meson.build:329]: @myungjoo mlagent provides pkgconfig thus ``` ml_agentd_dep = dependency('ml-agentd', required: false)``` is enough for most cases I guess. The other lines are only for providing fallback mechanisms to support the cases somehow the dependency method could not resolve although ```true``` (or ```auto```) has been given.

**Review by myungjoo — APPROVED** [review]

**Comment by myungjoo** [comment]
@wooksong Do you need further discussion? or can we merge this?

**Comment by wooksong** [comment]
> @wooksong Do you need further discussion? or can we merge this?

After resolving the conflIct, I will merge it. Thanks.

**Pull request merged**